chainreg/chainregistry.go

```diff
@@ -486,92 +485,11 @@ func NewPartialChainControl(cfg *Config) (*PartialChainControl, func(), error) {
 
 		// The api we will use for our health check depends on the
 		// bitcoind version.
-		cmd, ver, err := getBitcoindHealthCheckCmd(chainConn)
+		cmd, err := getBitcoindHealthCheckCmd(chainConn)
 		if err != nil {
 			return nil, nil, err
 		}
 
-		// If the getzmqnotifications api is available (was added in
-		// version 0.17.0) we make sure lnd subscribes to the correct
-		// zmq events. We do this to avoid a situation in which we are
-		// not notified of new transactions or blocks.
-		if ver >= 170000 {
-			zmqPubRawBlockURL, err := url.Parse(bitcoindMode.ZMQPubRawBlock)
-			if err != nil {
-				return nil, nil, err
-			}
-			zmqPubRawTxURL, err := url.Parse(bitcoindMode.ZMQPubRawTx)
-			if err != nil {
-				return nil, nil, err
-			}
-
-			// Fetch all active zmq notifications from the bitcoind client.
-			resp, err := chainConn.RawRequest("getzmqnotifications", nil)
-			if err != nil {
-				return nil, nil, err
-			}
-
-			zmq := []struct {
-				Type    string `json:"type"`
-				Address string `json:"address"`
-			}{}
-
-			if err = json.Unmarshal([]byte(resp), &zmq); err != nil {
-				return nil, nil, err
-			}
-
-			pubRawBlockActive := false
-			pubRawTxActive := false
-
-			for i := range zmq {
-				if zmq[i].Type == "pubrawblock" {
-					url, err := url.Parse(zmq[i].Address)
-					if err != nil {
-						return nil, nil, err
-					}
-					if url.Port() != zmqPubRawBlockURL.Port() {
-						return nil, nil, fmt.Errorf(
-							"unable to subscribe to zmq block events on "+
-								"%s (bitcoind is running on %s)",
-							zmqPubRawBlockURL.Host,
-							url.Host,
-						)
-					}
-					pubRawBlockActive = true
-				}
-				if zmq[i].Type == "pubrawtx" {
-					url, err := url.Parse(zmq[i].Address)
-					if err != nil {
-						return nil, nil, err
-					}
-					if url.Port() != zmqPubRawTxURL.Port() {
-						return nil, nil, fmt.Errorf(
```

Comment:
How about we just log a WRN message here? I think this could still be valuable for users that have a mistake in their configuration.

Reply:
@guggero Good idea. Code updated.